src/main/java/com/example/pokebattlez/battle/model/BattlePokemon.java

```diff
@@ -52,6 +72,16 @@ public void setNature(Nature nature) {
         stats.get(nature.getDown()).modifyNature(-NATURE_MODIFIER);
     }
 
+    public void setPokemonService(PokemonService pokemonService) {
+        this.pokemonService = pokemonService;
+        pokemonService.getApiData(this);
+        currentHp = stats.get(HP).getValue();
+    }
+
+    public void registerMove(Move move) {
```

Comment:
Why move registration is on Pokemon and not on the Battle?

Reply:
Each pokemon can learn up to 4 moves, move is just the name for attacks.